docs/docs/development/hardware-integration/new-board.md

```diff
@@ -0,0 +1,181 @@
+board:
+  ...
+  socs:
+  - name: <soc-1>
+    variants:
+    - name: zmk
+```
+
+### Add Kconfig for ZMK Variant
+
+Add a file to your board's folder called `<your-board>_zmk_defconfig`. This file will be used to set Kconfig flags specific to the ZMK variant. These flags are typically a subset of the following:
+
+```yaml title="<your-board>_zmk_defconfig"
+# SPDX-License-Identifier: <your license>
+
+# Enable MPU
+CONFIG_ARM_MPU=y
+
+CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC=125000000
+
+# Enable reset by default
+CONFIG_RESET=y
+
+# Enable clock control by default
+CONFIG_CLOCK_CONTROL=y
+
+# Code partition needed to target the correct flash range
+CONFIG_USE_DT_CODE_PARTITION=y
+
+# Output UF2 by default, native bootloader supports it.
+CONFIG_BUILD_OUTPUT_UF2=y
+
+# USB HID
+CONFIG_ZMK_USB=y
+
+# BLE HID
+CONFIG_ZMK_BLE=y
+
+# Bootloader Support
+CONFIG_RETAINED_MEM=y
+CONFIG_RETENTION=y
+CONFIG_RETENTION_BOOT_MODE=y
+
+# Settings Support
+CONFIG_MPU_ALLOW_FLASH_WRITE=y
+CONFIG_NVS=y
+CONFIG_SETTINGS_NVS=y
+CONFIG_FLASH=y
+CONFIG_FLASH_PAGE_LAYOUT=y
+CONFIG_FLASH_MAP=y
+
+
+# Enable HW stack protection
+CONFIG_HW_STACK_PROTECTION=y
+
+# Enable GPIO
+CONFIG_GPIO=y
+
+# Defaults for matrix scanning to avoid interrupt issues
+CONFIG_ZMK_KSCAN_MATRIX_POLLING=y
+```
+
+Note that none of our in-tree boards have all of the above flags set. We recommend copying the Kconfig flags from an existing in-tree board with the same SoC as the one you are using, rather than figuring out which ones you need yourself.
+
+### Add Devicetree for ZMK Variant
+
+```dts title="<your-board>_zmk.dts
+/*
+ * Copyright (c) <year> <you>
+ *
+ * SPDX-License-Identifier: <your license>
+ */
+
+//include the base board definition
+#include <../boards/<vendor or you>/<board folder name>/<your board>.dts>
+//include predefined boot mode settings, e.g.
+#include <arm/raspberrypi/rp2040-boot-mode-retention.dtsi>
+
+//disable UART nodes if they are not actively in use (wired split) as they increase power draw
+&uart0 { status = "disabled"; };
+
+//reduce the code partition section of the memory and add a storage partition to store ZMK Studio changes
+&code_partition {
+    reg = <0x100 (DT_SIZE_M(2) - 0x100 - DT_SIZE_K(512))>;
+};
+&flash0 {
+    reg = <0x10000000 DT_SIZE_M(2)>;
+    partitions {
+        storage_partition: partition@180000 {
+            reg = <0x180000 DT_SIZE_K(512)>;
+            read-only;
+        };
+    };
+};
+```
+
+See [here](./bootloader/index.mdx) for bootloader instructions for other SoCs. Depending on your SoC and design you may need to make further changes. Please refer to our in-tree boards with the same SoC as yours as examples.
+
+## Next Steps
+
+If your board is a keyboard, continue from ["Kconfig.defconfig"](https://zmk.dev/docs/development/hardware-integration/new-shield?keyboard-type=unibody#kconfigdefconfig) of the new shield guide. Use your ZMK variant's devicetree instead of the overlay file which would be used for a shield.
```

Comment:
```suggestion
If your board is a keyboard, continue from [`Kconfig.defconfig`](https://zmk.dev/docs/development/hardware-integration/new-shield?keyboard-type=unibody#kconfigdefconfig) step of the new shield guide. Use your ZMK variant's devicetree instead of the overlay file which would be used for a shield.
```